src/pyhf/readxml.py

```diff
@@ -343,20 +367,41 @@ def dedupe_parameters(parameters):
     return list({v['name']: v for v in parameters}.values())
 
 
-def parse(configfile, rootdir, track_progress=False, validation_as_error=True):
+def parse(
+    configfile: PathOrStr | T.IO[bytes] | T.IO[str],
+    rootdir: PathOrStr,
+    mounts: T.Optional[MountPathType] = None,
+    track_progress: bool = False,
+    validation_as_error: bool = True,
+):
+    """
+    Parse the configfile with respect to the rootdir.
```

Comment:
```suggestion
    Parse the ``configfile`` with respect to the ``rootdir``.
```

formatting suggestion for the docs